File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0443_String_Compression.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0443_String_Compression {
    public int compress(char[] chars) {
        int i = 0, cnt = 1;
        for (int j = 0; j < chars.length; j++) {
            if (j + 1 < chars.length && chars[j] == chars[j + 1]) {
                cnt++;
            } else {
                chars[i++] = chars[j];
                if (cnt > 1) {
                    String count = "" + cnt;
                    for (int k = 0; k < count.length(); k++) {
                        chars[i++] = count.charAt(k);
                    }
                }
                cnt = 1;
            }
        }
        return i;
    }
}

File: algorithms/algorithms-cplusplus/leetcode/Question_0443_String_Compression.cc
#include <string>
#include <vector>

using namespace std;

class Solution {
 public:
  int compress(vector<char>& chars) {
    int i = 0, cnt = 1, n = chars.size();
    for (int j = 0; j < n; j++) {
      if (j + 1 < n && chars[j] == chars[j + 1]) {
        cnt++;
      } else {
        chars[i++] = chars[j];
        if (cnt > 1) {
          string count = to_string(cnt);
          for (char c : count) {
            chars[i++] = c;
          }
        }
        cnt = 1;
      }
    }
    return i;
  }
};
